orchestrator/services/processes.py

```diff
@@ -155,54 +138,101 @@ def _db_log_step(
             else:
                 p.assignee = Assignee.SYSTEM
 
-        # check if last error state is identical to determine if we add a new step or update the last one
-        last_db_step = p.steps[-1] if len(p.steps) else None
+    else:
+        p.failed_reason = None
+        p.traceback = None
+
+    return p
+
 
+def _get_current_step_to_update(
+    stat: ProcessStat, p: ProcessTable, step: Step, process_state: WFProcess
+) -> ProcessStepTable:
+    """Checks if last error state is identical to determine if we add a new step or update the last one."""
+    step_state: State = process_state.unwrap()
+    current_step = None
+    last_db_step = p.steps[-1] if len(p.steps) else None
+    current_sub_step, current_step_group = step_state.get("__sub_step"), step_state.get("__step_group")
+    db_sub_step, db_step_group = (
+        (last_db_step.state.get("__sub_step"), last_db_step.state.get("__step_group")) if last_db_step else (None, None)
+    )
+
+    if current_sub_step is None and db_sub_step is None:
+        pass  # Skip step group logic
+    elif (db_sub_step is None and current_sub_step) or (db_sub_step and current_sub_step is None):
```

Comment:
Looks like a xor? Maybe you can use `elif db_sub_step ^ current_sub_step:`

Reply:
Oof. True but less readable I thbink